homeworks/olena.ambitska_elena-ambitska/GamesProject/scripts/games.js

```diff
@@ -0,0 +1,278 @@
+                            </div>
+
+                            <div>
+                                <dt>Publisher</dt>
+                                <dd>${game.publisher}</dd>
+                            </div>
+
+                            <div>
+                                <dt>Developer</dt>
+                                <dd>${game.developer}</dd>
+                            </div>
+
+                            <div>
+                                <dt>Genre</dt>
+                                <dd>${game.genre}</dd>
+                            </div>
+
+                            <div>
+                                <dt>Release_date</dt>
+                                <dd >${game.release_date}</dd>
+                            </div>
+                        </dl>
+                    </li>`;
+    }
+
+    function renderCards(container) {
+        const cardsHTML = games.map(createCardTemplate).join('');
+        container.querySelector('.section__title').insertAdjacentHTML('afterend', `<ul class="card__list">${cardsHTML}</ul>`);
+    }
+
+    function updateState() {
+        filtersState.checkboxes = [...checkboxes].filter((checkbox) => checkbox.checked);
+        filtersState.selectedGenre = genreDropdown.options[genreDropdown.selectedIndex].value;
+        filtersState.textValue = textInput.value.toLowerCase();
+        filtersState.selectedRadio = [...radioButtons].find((radio) => radio.checked).value;
+    }
+
+    function applyRadioFilter(cardItem) {
+        const textContent = cardItem.textContent.toLowerCase();
+        const platform = (cardItem.dataset.platform).toLowerCase();
+
+        const isPlatformVisible = filtersState.selectedRadio === 'platform' && platform.includes('pc (windows)');
+        const isOnlineVisible = filtersState.selectedRadio === 'online' && textContent.includes('web browser');
+        const isAllVisible = filtersState.selectedRadio === 'all';
+
+        return isPlatformVisible || isOnlineVisible || isAllVisible;
+    }
+
+    function applyCheckboxFilter(cardItem) {
+        const releaseDate = new Date(cardItem.dataset.releaseDate).getFullYear();
+        const isOld = releaseDate < 2010;
+        const isNew = releaseDate > 2020;
+
+        return (filtersState.checkboxes.length === 0
+            || filtersState.checkboxes.some(({ value }) => (value === 'new' && isNew)
+                || (value === 'old' && isOld)));
+    }
+
+    function applyTextSearchFilter(cardItem) {
+        const textContent = cardItem.textContent.toLowerCase();
+
+        return textContent.includes(filtersState.textValue);
+    }
+
+    function applyDropdownFilter(cardItem) {
+        const genre = (cardItem.dataset.genre).toLowerCase();
+
+        return (filtersState.selectedGenre === 'all' || genre === filtersState.selectedGenre);
+    }
+
+    function updateVisibility(cardItem) {
+        const visibleRadioFilter = applyRadioFilter(cardItem);
```

Comment:
Common rules:
- if a variable contains `boolean` values, its name starts from `is` or `has`
- if a function returns something, its name starts from `get`
```suggestion
        const isVisibleByPlatform = getPltformVisibility(cardItem);
        ...
```

BTW `updateVisibility(cardItem)` - cool name

Reply:
You could have one more radio filter in the future, so for naming better use the nature (platform), rather than the form (radio)

Reply:
done. Thank you for your comments. Very useful.